tests/integration/targets/ome_application_network_proxy/tests/set.yaml

```diff
@@ -0,0 +1,96 @@
+# Dell OpenManage Ansible modules
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- name: Test that we have an OME host, OME username and OME password
+  ansible.builtin.fail:
+    msg: 'Please define the following variables: hostname, username
+         and password.'
+  when: hostname is not defined or username is not defined or password
+         is not defined
+
+- block:
+    # PRE REQ - CONFIGURE NETWORK PROXY
+    - name: Reset proxy configuration.
+      ome_application_network_proxy:
+        enable_proxy: false
+
+    - name: Update proxy configuration and enable authentication. (check mode - changes expected)
+      ome_application_network_proxy: &configure_network_proxy
+        enable_proxy: true
+        ip_address: "192.168.0.2"
+        proxy_port: 444
+        enable_authentication: true
+        proxy_username: "proxy_username"
+        proxy_password: "proxy_password"
+      register: cm_changes
+      check_mode: true
+
+    # CONFIGURE NETWORK PROXY
+    - name: Update proxy configuration and enable authentication.
+      ome_application_network_proxy:
+        <<: *configure_network_proxy
+      register: configure_network_proxy
+
+    - name: Get application network time details by uri.
+      uri:
+        url: https://{{ hostname | ansible.utils.ipwrap }}:{{port}}/api/ApplicationService/Network/ProxyConfiguration
+        user: "{{ username }}"
+        password: "{{ password }}"
+        validate_certs: false
+        status_code: 200
+      register: uri_proxy
+
+    - name: Update proxy configuration and enable authentication.(check mode - no changes expected)
+      ome_application_network_proxy:
+        <<: *configure_network_proxy
+      register: cm_no_changes
+      check_mode: true
+
+    - name: Update proxy configuration and enable authentication.
+      ome_application_network_proxy:
+        <<: *configure_network_proxy
+      register: configure_network_proxy_idempotent
+
+    - name: Reset proxy authentication.
+      ome_application_network_proxy:
+        enable_proxy: true
+        ip_address: "192.168.0.2"
+        proxy_port: 444
+        enable_authentication: false
+      register: reset_network_proxy_authentication
+
+    - name: Reset proxy configuration.
+      ome_application_network_proxy:
+        enable_proxy: false
+      register: reset_network_proxy_config
+
+    - name: Verify the status of tasks for enable_ntp and set addresses
+      assert:
```

Comment:
Move all the assertion under respective task. Make this changes at all applicable places

Reply:
updated